README.md

```diff
@@ -1 +1,112 @@
-# Room Access rules
\ No newline at end of file
+# Room Access rules
+
+This module implements handling around the `im.vector.room.access_rules` state event, which spec is described below.
+
+## `im.vector.room.access_rules`
+
+Restricts the access to a room based on the selected preset. Body:
+
+```json
+{
+    "rule": "<rule>"
+}
+```
+
+`<rule>` is either `restricted`, `unrestricted` or `direct`.
+
+The implementation of the different presets lives in the
+`synapse.third_party_rules.access_rules` module.
+
+### `restricted` preset
+
+Default preset for non-direct rooms.
+
+Forbids any invite and membership update for users that belong to a server
+that is in the blacklist provided by the server's configuration
+(`domains_forbidden_when_restricted`). If the invite is a 3PID invite, queries
+a custom `/info` endpoint of the configured identity server to check if that email
+address would belong to a blacklisted server.
+
+### `unrestricted` preset
+
+Doesn't apply any restriction on who can join the room.
+
+Forbids any `m.room.power_levels` event that either:
+
+* change the default power level to a non-0 value, or
+* change the power level for a user from a blacklisted server (see details about the `restricted` preset) to a non-default value
+
+### `direct` preset
+
+Default preset for direct rooms (i.e. rooms created with `"direct": true`).
+
+Only allow two members in the room by running the following algorithm for
+each new event of type `m.room.member` or `m.room.third_party_invite` sent
+into the room:
+
+0. retrieve the list of memberships and 3PID invite tokens from the room's state, which in practice means retrieving the state key of every `m.room.member` or `m.room.third_party_invite` event present in the room's state (ignoring 3PID invite events with an empty content)
+
+1. if there already is a `m.room.third_party_invite` event in the room's state and the new event is of the same type, refuse the event if the state key isn't the same as used by one of the 3PID invites returned by step 0
```

Comment:
```suggestion
1. if the event is of type `m.room.third_party_invite`, reject it if an existing `m.room.third_party_invite` event in the room has the same state key
```

Reply:
Err, no, that's not true. It rejects the event if there's at least one other 3pid invite in the room, but none of the 3pid invites in the room's state has the same state key as the new event.

```suggestion
1. if the event is of type `m.room.third_party_invite`, and there are already events of the same type in the room's state, reject the new event if its state key doesn't match the state key of one of the existing events.
```